src/js/popup/popup.js

```diff
@@ -1,802 +1,1350 @@
-      "imgSrc": "announcements/panel-announcement-password-manager-relay-illustration.svg",
-      "imgSrcPremium": "announcements/panel-announcement-password-manager-relay-square-illustration.svg",
-      "tipHeadline": browser.i18n.getMessage("popupPasswordManagerRelayHeadline"),
-      "tipBody": browser.i18n.getMessage("popupPasswordManagerRelayBody")
-    }
-  }
-
-  return panels;
-}
-
-async function getOnboardingPanels() {
-  // Conditions for showing the Firefox password manager announcement
-  const isFirefoxIntegration = await checkWaffleFlag("firefox_integration");
-
-  const panels = {
-    "announcements": {
-      "panel1": {
-        "imgSrc": "announcements/panel-announcement-attachment-limit.svg",
-        "tipHeadline": browser.i18n.getMessage("popupAttachmentSizeIncreaseHeadline"),
-        "tipBody": browser.i18n.getMessage("popupAttachmentSizeIncreaseBody"),
+      dismissErrorClick: async (e) => {
+        e.preventDefault();
+        e.target.classList.remove("is-shown");
       },
-      "panel2": {
-        "imgSrc": "announcements/panel-announcement-critical-emails.svg",
-        "tipHeadline": browser.i18n.getMessage("popupBlockPromotionalEmailsHeadline_2"),
-        "tipBody": browser.i18n.getMessage("popupBlockPromotionalEmailsBodyNonPremium"),
+      externalClick: async (e) => {
+        e.preventDefault();
+        if (e.target.dataset.eventLabel && e.target.dataset.eventAction) {
+          sendRelayEvent(
+            "Panel",
+            e.target.dataset.eventAction,
+            e.target.dataset.eventLabel
+          );
+        }
+        await browser.tabs.create({ url: e.target.href });
+        window.close();
       },
-      "panel3": {
-        "imgSrc": "announcements/panel-announcement-sign-back-in.svg",
-        "tipHeadline": browser.i18n.getMessage("popupSignBackInHeadline_mask"),
-        "tipBody": browser.i18n.getMessage("popupSignBackInBody_mask_v2"),
+      navigationClick: (e) => {
+        e.preventDefault();
+        document
+          .querySelector(".js-internal-link.is-active")
+          ?.classList.remove("is-active");
+        e.target.classList.add("is-active");
+        const panelId = e.target.dataset.panelId;
+        popup.panel.update(panelId);
       },
-    },
-    "maxAliasesPanel": {
-      "imgSrc": "high-five.svg",
-      "tipHeadline": browser.i18n.getMessage("popupOnboardingMaxAliasesPanelHeadline"),
-      "tipBody": browser.i18n.getMessage("popupOnboardingMaxAliasesPanelBody"),
-      "upgradeButton": browser.i18n.getMessage("popupUpgradeToPremiumBanner"),
-      "upgradeButtonIcon": "/icons/icon.svg",
-    },
-    "premiumPanel": {
-      "aliasesUsedText": browser.i18n.getMessage("popupAliasesUsed_mask"),
-      "emailsBlockedText": browser.i18n.getMessage("popupEmailsBlocked"),
-      "emailsForwardedText": browser.i18n.getMessage("popupEmailsForwarded"),
-    }
-  };
+      generateMask: async (event, type = "random", data = null) => {
```

Comment:
Not worth a big refactor, but if it's easy to change, would be great to use named parameters instead of positional ones for optional parameters (specifically `type`), to make it easier to understand at the calling site.